File: sololearn/US2EUDate/US_To_EU_Date.py
# US date (input)
us_date = str(input())

# input case 1 (11/19/2019)
if len(us_date) <= 10:
    us_date = us_date.split('/')
# input case 2 (November 19, 2019)
else:
    us_date = us_date.split()
    us_date[1] = us_date[1].split(',')[0]

# date
date = us_date[1]

# month
if len(us_date[0]) > 2:
    # months array
    months = ["NULL", "January", "Febuary", "March", "April", "May", "June", "July", "August", "September", "October", "November", "December"]
    month = str(months.index(us_date[0]))
else:
    month = us_date[0]

# year 
year = us_date[2]

# EU Date
print(date + "/" + month + "/" + year)

File: sololearn/US2EUDate/US_To_EU_Date.cpp
/*
Author: chankruze (chankruze@geekofia.in)
Created: Sun Aug 09 2020 18:37:43 GMT+0530 (India Standard Time)

Copyright (c) Geekofia 2020 and beyond

*/

#include <bits/stdc++.h>

#include <algorithm>
#include <iostream>
#include <string>
#include <vector>

using namespace std;

int main(int argc, char const *argv[]) {
    int date, month, year;
    string part;

    // US date (input)
    string usDate;
    getline(cin >> ws, usDate);

    stringstream ss(usDate);

    if (usDate.length() <= 10) {
        vector<int> iParts;

        while (getline(ss >> ws, part, '/')) {
            iParts.emplace_back(stoi(part));
        }

        date = iParts[1];
        month = iParts[0];
        year = iParts[2];
    } else {
        vector<string> sParts;

        while (getline(ss >> ws, part, ' ')) {
            sParts.emplace_back(part);
        }

        // date
        date = stoi(sParts[1].substr(0, sParts[1].find(',')));

        // month
        vector<string> months = {"NULL", "January", "Febuary", "March", "April", "May", "June", "July", "August", "September", "October", "November", "December"};
        month = distance(months.begin(),
                         find(months.begin(), months.end(), sParts[0]));

        // year
        year = stoi(sParts[2]);
    }

    cout << date << "/" << month << "/" << year << endl;
    return 0;
}
